lib/CSkillHandler.cpp

```diff
@@ -45,7 +45,7 @@ int32_t CSkill::getIndex() const
 
 int32_t CSkill::getIconIndex() const
 {
-	return getIndex(); //TODO: actual value with skill level
+	return getIndex() * 3 + 3; //TODO: actual value with skill level
```

Comment:
Tried to fix, but still doesn't return index depending on level.

Reply:
One option would be to add an override. 
Check getNameSingular / getNamePlural in CCreature.

So, CSkill will have private `getIconIndex()` override that returns... well any arbitrary-selected level (basic?)
And public getIconIndex(int skillMasterLevel) that returns icon for specified mastery.

So all code that calls Entity::getIconIndex() will still work and get some icon, and all code that is aware that this is a skill will need to specify mastery.